### DISCUSSION TITLE
Segmentation fault  (core dumped) Error in dosxyznrc simulation using source 21

### DISCUSSION THREAD

**Discussion by physicsmed** [nrc-cnrc/EGSnrc#1086] [Questions] [answered]
Hello everyone,
i am trying to run a simulation in parallel mode with 8 CPU under a cluster, using a step and shoot mode and both MLC and Jaws were synchronized using sequence files.
the phantom used is egsphant file obtained using ctcreate. the problem is when i run with 4 cpu the simulation works succesfully, but when i try to use 8 cpu the simulation creates the files inputfile_w1.egslog/egslst/errors/pardose and the same for w2 until w8. but the w5 files are not completed as you see in the picture below, and i think the problem is due to it. it means the simulation were reached its final steps to be finished and it causes the problem of core dumped. 
NB: i have used the same number of histories in the both run (4 spu and 8 cpu). 
![image psd(38)](https://github.com/nrc-cnrc/EGSnrc/assets/134697172/f5c25d3d-a6f7-4ce5-9dd7-fbf4b9ba124a)

**Comment by rtownson** [comment]
You can add together your results without the _w5 ones - just rename your 6,7,8 to 5,6,7 and then use the "combine" mode to add the results together.

You could take a look in the egsrun directory to see if there are any helpful errors, and trying running in "restart" mode.

Or, just run 1/8th the number of histories, rename to _w5, and then use "combine" to add them together.

Around the root cause - do you use a boundary material around the .egsphant? Make sure it is set to air and not vacuum, and fully encloses the source location. This may require editing the .egsinp file directly, let me know if you need help with this.

**Reply by physicsmed** [reply]
Thank u for replying Sir and sorry for the late response, I still encountering the same problem, but i am trying  to change medium surrounding the phantom using DOSXYZnrc GUI, it visualize just the choice of vacuum, there is no air. do i have to change it from the input file by changing any parameter? 
NB: i have dsource =40 cm and dsurrounding = 30cm, can this cause the problem?

**Reply by rtownson** [reply, marked as answer]
Yes it is worth trying increasing dsurround=41cm.

To change it to air instead of vacuum, open your .egsinp file in a text editor. Look for the "41", it will be near the bottom of the numbers, maybe the 3rd to last line (but before the transport parameters). It probably looks like this:
```
2, 0, 0, 41, 0, 0, 0, 0
```
Change the number before the 41 to be the index of air in your .egsphant file. Your .egsphant file has a list of materials at the top of the file when you open it in a text editor, if the first material in the list is air, then you want to put "1" (0 is for vacuum). E.g.

```
2, 0, 1, 41, 0, 0, 0, 0
```
Save the input file and try the simulation again.

Note that this input line is described on page 29 of the DOSXYZnrc user manual.

**Reply by physicsmed** [reply]
Thank you, it has worked successfully.
